extensions/2.0/Khronos/KHR_materials_iridescence/README.md

```diff
@@ -0,0 +1,336 @@
+</figure>
+
+The thin-film layer can have a different IOR than the underlying material. With `iridescenceIOR` one can set an IOR value for the thin-film layer independently.
+This also has an effect on the optical path difference:
+
+<figure>
+<img src="./figures/ior-comparison.jpg"/>
+<figcaption><em>Comparison of different iridescence IOR values for a thin-film thickness range between 200 nm (bottom) and 800 nm (top).</em></figcaption>
+</figure>
+
+The iridescence effect is modeled via a microfacet BRDF with a modified Fresnel reflectance term that accounts for inter-reflections as shown in [Laurent Belcour and Pascal Barla (2017)](https://belcour.github.io/blog/research/publication/2017/05/01/brdf-thin-film.html).
+
+### Clearcoat
+
+Considering `KHR_materials_clearcoat` extension, the clearcoat layer is evaluated before the thin-film layer.
+Light that is passing through the clearcoat is then processed as described taking into account differences in IOR.
+
+<figure>
+<img width=40% src="./figures/layering.png"/>
+<figcaption><em>When adding clearcoat ontop of the material, the interface between the "outside" and the thin-film changes and the IOR of the clearcoat (usually 1.5) has to be used for the outside IOR.</em></figcaption>
+</figure>
+
+## Implementation Notes
+
+*This section is non-normative.*
+
+For the calculation of `f_specular`, the original Fresnel term `F_schlick` (see [B.3.4. Fresnel](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#fresnel) in the glTF 2.0 specification) is exchanged with a newly introduced `F_iridescence` term.
+To artistically control the strength of the iridescence effect, the two Fresnel terms are mixed by using the `iridescenceFactor`:
+
+```glsl
+F = mix(F_schlick, F_iridescence, iridescenceFactor)
+```
+
+The calculation of `F_iridescence` is described in the following sections. For glTF an approximation of the original model (defined in the Mitsuba code example from [Belcour/Barla](https://belcour.github.io/blog/research/publication/2017/05/01/brdf-thin-film.html) is used.
+
+### Material Interfaces
+
+The iridescence model defined by Barla and Belour models two material interfaces - one from the outside to the thin-film layer and another one from the thin-film to the base material. These two interfaces are defined as follows:
+
+```glsl
+// First interface
+float R0 = IorToFresnel0(iridescenceIOR, outsideIOR);
+float R12 = F_Schlick(R0, cosTheta1);
+float R21 = R12;
+float T121 = 1.0 - R12;
+
+// Second interface
+vec3 baseIOR = Fresnel0ToIor(baseF0 + 0.0001); // guard against 1.0
+vec3 R1 = IorToFresnel0(baseIOR, iridescenceIOR);
+vec3 R23 = F_Schlick(R1, cosTheta2);
+```
+
+`iridescenceIOR` is the index of refraction of the thin-film layer, `outsideIOR` the IOR outside the thin-film (e.g. air or the clearcoat material) and `baseIOR` the IOR of the base material. The latter is calculated from its F0 value using the inverse of the special normal incidence case of Fresnel's equations:
```

Comment:
> `outsideIOR` the IOR outside the thin-film (e.g. air or the clearcoat material) and `baseIOR` the IOR of the base material

This could be more specific. In which case is the clearcoat IOR used exactly? When `KHR_materials_clearcoat` is present? When `clearcoatFactor > 0`? Always, even if clearcoat is disabled (`clearcoat == 0`)? Maybe the answer should go into the normative part.

Reply:
I think that's still up to debate. The current implementation in the sample viewer is, that we blend between an IOR of `1.0` (air) and `1.5` (clearcoat IOR) using the `clearcoatFactor`. That's just a workaround and not physically plausible, but also weighting the clearcoat effect using a factor is not physically based. Another solution for clearcoat would be to ignore it for this extension. Artistically this might be a good approach as it wouldn't mess with an existing iridescence effect when adding clearcoat but it would again contradict the iridescence model.

Reply:
The clearcoat dependency has been removed as discussed in the PBR TSG meeting.